### DISCUSSION TITLE
Has segment-geospatial finetuned SAM on a remote sensing dataset?

### DISCUSSION THREAD

**Discussion by adamschneider42** [opengeos/segment-geospatial#103] [Q&A] [answered]
I am just wondering if the SAM has been finetuned on a remote sensing dataset, or the code has just been modified to handle geotiffs and geospatial data?

**Comment by giswqs** [comment, marked as answer]
Not yet, but contributions are welcome. See https://github.com/opengeos/segment-geospatial/issues/35

**Comment by amalshehan** [comment]
> For now, we can only fine-tune what SAM allows. See the parameters below. It might take sometime before we can support other model fine-tuning. Contributions are always welcome.
> 
> https://github.com/opengeos/segment-geospatial/blob/main/samgeo/samgeo.py#L43
> 
> ```python
>   points_per_side: Optional[int] = 32,
>   points_per_batch: int = 64,
>   pred_iou_thresh: float = 0.88,
>   stability_score_thresh: float = 0.95,
>   stability_score_offset: float = 1.0,
>   box_nms_thresh: float = 0.7,
>   crop_n_layers: int = 0,
>   crop_nms_thresh: float = 0.7,
>   crop_overlap_ratio: float = 512 / 1500,
>   crop_n_points_downscale_factor: int = 1,
>   point_grids: Optional[List[np.ndarray]] = None,
>   min_mask_region_area: int = 0,
>   output_mode: str = "binary_mask",
> ```

Any examples of tuning with these parameters? samgeo seems to be unable to pass the parameters to sam. Has anyone else had a similar issue ? It is throwing a unexpected keyword argument error!

**Reply by amalshehan** [reply]
Figured out the issue: For automatic segmentation, it works, but for SamPredictor it does not take the parameters in the __init__ although samgeo gives and option to pass **kwargs to it. 

samgeo:
https://github.com/opengeos/segment-geospatial/blob/main/samgeo/samgeo.py#L104

segment-anything:
https://github.com/facebookresearch/segment-anything/blob/main/segment_anything/predictor.py#L18

**Reply by giswqs** [reply]
So is this a samgeo issue or the original SAM issue?  samgeo already has `sam_kwargs` in the `SamPredictor` constructor. 

```python
if automatic:
    # Segment the entire image using the automatic mask generator
    self.mask_generator = SamAutomaticMaskGenerator(self.sam, **sam_kwargs)
else:
    # Segment selected objects using input prompts
    self.predictor = SamPredictor(self.sam, **sam_kwargs)
```

**Reply by amalshehan** [reply]
The issue is on both sides. In the original SAM SamPredictor does not take kwargs so better to remove the **sam_kwargs from the call to it in samgeo.

**Reply by giswqs** [reply]
It appears to be the issue of the original SAM SamPredictor. I will keep `sam_kwargs` in the samgeo function in case SAM supports it in the future.

**Reply by amalshehan** [reply]
@giswqs Ok thanks for looking into this. If possible better to update the geosam docs. As you suggest good to keep it since it is vital to have that option when using it on diverse types of data. Is there a workaround to this? I recall ArcPro deployment allowing parameter tuning!

**Reply by amalshehan** [reply]
@giswqs any plans to improve samgeo based on [sam-hq](https://github.com/SysCV/sam-hq)?
